modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/ClassifyByRangeProcess.java

```diff
@@ -0,0 +1,144 @@
+            Definition classify = new Definition();
+            if(outputAttribute == null || outputAttribute.trim().isEmpty()) {
+                outputAttribute = "class";
+            }
+            classify.name = outputAttribute;
+            try {
+                classify.binding = outputType != null ? Class.forName(outputType) : String.class;
+            } catch (ClassNotFoundException e) {
+                throw new ProcessException(outputType + " is not a valid value for outputType: should be a class name");
+            }
+
+            AttributeDescriptor classifyDescriptor = features.getSchema().getDescriptor(classifyOnAttribute);
+            if(classifyDescriptor == null) {
+                throw new ProcessException("classifyOnAttribute is not a valid schema attribute: " + classifyOnAttribute);
+            }
+            Class<?> rangeType = classifyDescriptor.getType().getBinding();
+            if(rangeType == null) {
+                rangeType = Number.class;
+            }
+            List<Expression> params = new ArrayList<Expression>();
+            params.add(filterFactory.property(classifyOnAttribute));
+            if((thresholds == null || thresholds.trim().isEmpty()) && (classifier == null || classifier.trim().isEmpty())) {
+                throw new ProcessException("at least one of thresholds and classifier is mandatory");
+            }
+            List<Object> ranges = new ArrayList<Object>();
+            if(classifier!= null && !classifier.trim().isEmpty()) {
+                if (classes == null) {
+                    classes = 5;
+                }
+                Function classifyFun = filterFactory.function(classifier, filterFactory.property(classifyOnAttribute), 
+                        filterFactory.literal(classes));
+                RangedClassifier rc = (RangedClassifier)classifyFun.evaluate(features);
+                for (int i = 0; i < rc.getSize(); i++) {
+                    ranges.add(rc.getMin(i));
+                }
+                ranges.add(rc.getMax(rc.getSize() - 1));
+            } else {
+                for(String threshold : thresholds.split(",")) {
+                    ranges.add(threshold);
+                }
+            }
+                
+            if(outputValues == null || outputValues.trim().isEmpty()) {
+                outputValues = "";
+                for(int count = 1; count <= ranges.size();count++) {
+                    outputValues += count + ",";
+                }
+                outputValues += (ranges.size() + 1 + "");
+            }
+            String[] rangesValues = outputValues.split(",");
+            if (rangesValues.length != (ranges.size() + 1)) {
+                throw new ProcessException("values are not consistent with thresholds (should be +1)");
+            }
+            for(int count = 0; count < ranges.size(); count++) {
+                params.add(filterFactory.literal(Converters.convert(rangesValues[count], classify.binding)));
```

Comment:
The converters will return null if the value cannot be converted, I guess we'd want an exception with a suitable error message if that happens?